app/context/authContext.tsx

```diff
@@ -24,26 +27,55 @@ export const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) =>
     },
   });
 
-  const loginMutation = useMutation<LoginResponse, Error, LoginParams>({
-    mutationFn: loginApi,
-    onSuccess: async (data) => {
-      // eslint-disable-next-line camelcase
-      const { user, access_token } = data.data;
+  // This is loginMutation function with next-auth
+  const loginMutation = useMutation({
+    mutationFn: async (formValues: LoginParams) => {
+      const { email, password, cfTurnstileResponse } = formValues;
 
-      saveToLocalStorage('user', user);
-      queryClient.setQueryData(['user'], user);
+      // Call NextAuth's signIn method
+      const result = await signIn('credentials', {
+        redirect: false,
+        email,
+        password,
+        cfTurnstileResponse,
+      });
 
-      // eslint-disable-next-line camelcase
-      saveToLocalStorage('access_token', access_token);
-
-      // eslint-disable-next-line camelcase
-      saveToLocalStorage('session', { token: access_token, role: user.role });
+      if (!result?.ok) {
+        throw new Error(result?.error || 'Failed to login');
+      }
+      return result;
+    },
+    onSuccess: () => {
+      // Redirect to the dashboard on success
+      router.push('/content-panel/dashboard');
```

Comment:
can we make this a parametar that will pass? this way the login could be reused for admin-panel as well?

Reply:
Yes, i made changes and this url is now dynamic.